src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java

```diff
@@ -50,22 +50,33 @@ public boolean shouldApplyIf(boolean isInsert) {
    * specifically Document's primary id, {@code _id}.
    */
   protected static String validateUpdatePath(UpdateOperator oper, String path) {
-    if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(path)
-        && !(oper.apiName().equals("$set")
-            || oper.apiName().equals("$unset")
-            || oper.apiName().equals("$setOnInsert"))) {
-      throw ErrorCodeV1.UNSUPPORTED_UPDATE_FOR_VECTOR.toApiException("%s", oper.apiName());
-    }
-
-    if (DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD.equals(path)
-        && !(oper.apiName().equals("$set")
-            || oper.apiName().equals("$unset")
-            || oper.apiName().equals("$setOnInsert"))) {
-      throw ErrorCodeV1.UNSUPPORTED_UPDATE_FOR_VECTORIZE.toApiException("%s", oper.apiName());
-    }
-
-    if (DocumentConstants.Fields.DOC_ID.equals(path)) {
-      throw ErrorCodeV1.UNSUPPORTED_UPDATE_FOR_DOC_ID.toApiException("%s", oper.apiName());
+    switch (path) {
+      case DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD:
+        switch (oper.apiName()) {
```

Comment:
-1 oper is an ENUM can we use the enum values rather than constants

Reply:
Ah good point. Was blindly converting existing code. Will change.